crates/sp-domains/src/storage_proof.rs

```diff
@@ -0,0 +1,273 @@
+use crate::verification::VerificationError;
+use crate::{DomainId, OpaqueBundle, RuntimeId, RuntimeObject};
+use frame_support::storage::generator::StorageMap;
+use frame_support::Identity;
+use hash_db::Hasher;
+use parity_scale_codec::{Decode, Encode, FullCodec};
+use scale_info::TypeInfo;
+use sp_core::storage::StorageKey;
+use sp_core::H256;
+use sp_runtime::traits::{Block as BlockT, HashingFor, NumberFor};
+use sp_std::marker::PhantomData;
+use sp_std::vec::Vec;
+use sp_trie::{read_trie_value, LayoutV1, StorageProof};
+
+#[cfg(feature = "std")]
+use sc_client_api::ProofProvider;
+
+pub struct StorageProofVerifier<H: Hasher>(PhantomData<H>);
+
+impl<H: Hasher> StorageProofVerifier<H> {
+    pub fn verify_and_get_value<V: Decode>(
+        state_root: &H::Out,
+        proof: StorageProof,
+        key: StorageKey,
+    ) -> Result<V, VerificationError> {
+        let db = proof.into_memory_db::<H>();
+        let val = read_trie_value::<LayoutV1<H>, _>(&db, state_root, key.as_ref(), None, None)
+            .map_err(|_| VerificationError::InvalidProof)?
+            .ok_or(VerificationError::MissingValue)?;
+
+        let decoded = V::decode(&mut &val[..]).map_err(|_| VerificationError::FailedToDecode)?;
+
+        Ok(decoded)
+    }
+}
+
+/// This is a representation of actual `SuccessfulBundle` storage in pallet-domains.
+/// Any change in key or value there should be changed here accordingly.
+pub struct SuccessfulBundlesStorage;
+impl StorageMap<DomainId, Vec<H256>> for SuccessfulBundlesStorage {
```

Comment:
We have recently changed this in Timestamp PR which is in main to take storage keys from the runtime for the client side and introduced a StorageKey trait to use runtime for pallets that needs access. Please update this and other below to use that route instead